awx/api/views/__init__.py

```diff
@@ -506,11 +514,19 @@ class InstanceGroupInstanceList(InstanceGroupMembershipMixin, SubListAttachDetac
     relationship = "instances"
     search_fields = ('hostname',)
 
-    def is_valid_relation(self, parent, sub, created=False):
+    def is_valid_relation(self, sub, created=False):
```

Comment:
Even if you don't use it, you still need to keep `parent` in the arg list, because it's a position argument. I hope that clears out the test failures, and then this will be ready to go.